sklearn/metrics/scorer.py

```diff
@@ -67,6 +69,43 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        classes = getattr(estimator, "classes_", None)
+
+        # if classes_ parameter doesn't exist
+        if classes is None:
+            return self._kwargs
+
+        if self._pass_classes in signature(self._score_func).parameters:
+            # if labels passed as kwargs, return kwargs as is
+            if self._pass_classes in self._kwargs:
+                if classes is not None:
+                    # labels should be a subset of classes
+                    if not set(classes).issuperset(
+                                        set(self._kwargs[self._pass_classes])):
+                        raise ValueError("`estimator classes=%s` is not the"
+                                         "superset of `scorer %s=%s`" %
+                                         (classes,
+                                          self._pass_classes,
+                                          self._kwargs['labels']))
+                return self._kwargs
```

Comment:
not covered!

Reply:
oh yea.. this will happen in case the estimator is something other than a classifier. but then the scorer_func will not have a labels parameter. should I still check it just in case a user defined metrics for say regressor has a labels parameter?